README.md

```diff
@@ -63,40 +63,46 @@ An example of a simple randomized cocotb testbench:
 
 ```python
 # test_dff.py
-
 import random
 
 import cocotb
 from cocotb.clock import Clock
-from cocotb.triggers import RisingEdge
-from cocotb.types import LogicArray
+from cocotb.triggers import ReadOnly, RisingEdge
 
-@cocotb.test()
+@cocotb.test
 async def dff_simple_test(dut):
     """Test that d propagates to q"""
 
-    # Assert initial output is unknown
-    assert LogicArray(dut.q.value) == LogicArray("X")
     # Set initial input value to prevent it from floating
     dut.d.value = 0
 
     clock = Clock(dut.clk, 10, units="us")  # Create a 10us period clock on port clk
     # Start the clock. Start it low to avoid issues on the first RisingEdge
     cocotb.start_soon(clock.start(start_high=False))
 
-    # Synchronize with the clock. This will regisiter the initial `d` value
-    await RisingEdge(dut.clk)
-    expected_val = 0  # Matches initial input value
-    for i in range(10):
-        val = random.randint(0, 1)
-        dut.d.value = val  # Assign the random value val to the input port d
-        await RisingEdge(dut.clk)
-        assert dut.q.value == expected_val, f"output q was incorrect on the {i}th cycle"
-        expected_val = val # Save random value for next RisingEdge
-
-    # Check the final input on the next clock
+    # generate stimulus
+    stimulus = [random.randint(0, 1) for _ in range(10)]
+
+    # calculated expected outputs
+    expecteds = [val for val in stimulus]
```

Comment:
"expecteds" is a strange word, maybe just "inputs" (instead of stimulus) "outputs_expected"?

Either way, just copy input to output: `expecteds = stimulus`